File: src/dao/DAOInsumo.java
/**IFPB - Curso SI - Disciplina de POB
 * @author Prof Fausto Ayres
 */
package dao;

import java.util.List;

import com.db4o.query.Query;

import modelo.Funcionario;
import modelo.Insumo;

public class DAOInsumo extends DAO<Insumo>  {
	public Insumo readByNome (String nome){	
		Query q = manager.query();
		q.constrain(Insumo.class);
		q.descend("nome").constrain(nome);
		List<Insumo> resultados = q.execute();
		if (resultados.size()>0)
			return (Insumo) resultados.get(0);
		else
			return null;
	}
	
}

File: src/DAO/DAOInsumo.php
<?php
namespace projeto\DAO;
use \projeto\DAO\DAO;
use projeto\dados\Restaurante;

class DAOInsumo extends DAO {

    public function readByNome (String $nome) {
        $insumos = Restaurante::getInsumos();

        foreach ($insumos as $value) {
            if ($value->getNome()==$nome){
                return $value;
            } 
        }
        return null;
    }
}

?>